### PR TITLE
Corrected Dial Drift Behavior

### PR DESCRIPTION
**Pull request by newcomb-luke**
Now the dials can drift from any position in either direction, and keep track of their in-range zone.
The dials sound the alarm when they go out of range, and the user can reset them using the (for now hard-coded) 1-5

### CODE REVIEW

File: src/dial.rs [hunk 1]
```diff
@@ -23,27 +23,126 @@ const DIAL_NEEDLE_TICK_VALUE: f32 = 100.0;
...
+    pub start: f32,
+    pub end: f32,
+}
+
+impl DialRange {
+    pub fn new(start: f32, end: f32) -> Self {
+        Self { start, end }
+    }
+
+    pub fn contains(&self, value: f32) -> bool {
```
Comment (TroyNeubauer) [inline src/dial.rs:37]: nit: std::ops::Range might be a better way to represent this if it gets more complicated. This is fine as is

File: src/window.rs [hunk 2]
```diff
@@ -43,8 +44,14 @@ pub fn draw_gui() {
...
     let mut dials = Vec::new();
 
+    let range_size = 4000.0;
     for i in 0..num_dials {
-        let dial = Dial::new(i + 1, (i + 1) as f32 * 100.0);
+        let dial = Dial::new(
+            i + 1,
+            50.0,
+            DialRange::new(i as f32 * 200.0, i as f32 * 200.0 + range_size),
+            (i + 1).to_string().chars().next().unwrap(),
```
Comment (TroyNeubauer) [inline src/window.rs:53]: lol. If it works it works

File: src/window.rs [hunk 3]
```diff
@@ -195,7 +231,52 @@ pub fn draw_gui() {
                                 VirtualKeyCode::Right => {
                                     input_x.y = 0.0;
                                 }
-                                _ => {}
+                                k => {
```
Comment (TroyNeubauer) [inline src/window.rs:234]: nit: macro?

**Review by TroyNeubauer — APPROVED** [review]
Looks good overall. Lmk if you want me to write a declarative macro for the keypress stuff, but its not blocking for this PR

File: src/window.rs (hunk 2 above)
Comment (newcomb-luke) [reply src/window.rs:53]: This is just for until we read it from the configuration file haha

File: src/window.rs (hunk 3 above)
Comment (newcomb-luke) [reply src/window.rs:234]: I'm not sure how a macro would help us here, but we should definitely figure something out to clean it up in the future.

File: src/dial.rs (hunk 1 above)
Comment (newcomb-luke) [reply src/dial.rs:37]: I thought about it. I wanted to make sure it was a little more flexible but if we need any more methods on it that would be a good idea.

**Pull request merged**